docs_src/microservices/device/supported/device-onvif-camera/Walkthrough/deployment.md

```diff
@@ -241,8 +250,77 @@ Follow these instructions to update devices.
     [{"apiVersion":"v3","statusCode":201,"id":"fb5fb7f2-768b-4298-a916-d4779523c6b5"}]
     ```
     
-2. Map credentials using the `map-credentials.sh` script.  
+2. Map the credentials for the camera.
+
+!!! note
+    The preferred method for setting the credentials is through the use of the curl commands.
+
+=== "Curl Command"
+    !!! note
+        If running in secure mode all the api executions need the [JWT token](#token-generation) generated previously.
+
+    a. Check the status of the credentials map.
+    ```bash
+    curl -H "X-Consul-Token:<consul-token>" -X GET "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/AppCustom/CredentialsMap?keys=true" | jq .
+    ```
+    Example response:
+    ```bash
+    [
+    "edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/NoAuth",
+    "edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/credentials001",
+    "edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/credentials002"
+    ]
+    ```
+
+        
+
+    b. Enter your chosen credentials name and then execute the command to set the mapping name.
+
+    !!! note
+        To edit credentials, this command may be skipped, and then use one of the existing credentials from the previous response in the following commands.
+
+    ```bash
+    curl --data '<creds-name>' -H "X-Consul-Token:<consul-token>" -X PUT "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/Writable/InsecureSecrets/credentials001/SecretName"
+    ```
+    
+    c. Enter your chosen username and the name of the corresponding credentials, and then execute the command to set the username.
+    ```bash
+    curl --data '<username>' -H "X-Consul-Token:<consul-token>" -X PUT "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/Writable/InsecureSecrets/<creds-name>/SecretData/username"
+    ```
 
+    d. Enter your chosen password and the name of the corresponding credentials, and then execute the command to set the password.
+    ```bash
+    curl --data '<password>' -H "X-Consul-Token:<consul-token>"-X PUT 'http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/Writable/InsecureSecrets/<creds-name>/SecretData/password'"
+    ```
+
+    e. Enter your chosen authentication method and the name of the corresponding credentials, and then execute the command to set the authentication method.
+        
+    !!! note
+        The options are: `usernametoken`, `digest`, or `both`
+
+    ```bash
+    curl --data '<auth-method>' -H "X-Consul-Token:<consul-token>" -X PUT "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/Writable/InsecureSecrets/credentials001/SecretData/mode"
+    ```
+
+    f. Enter your chosen credentials name and then execute the command to create the mapping.
+    ```bash
+    curl -H "X-Consul-Token:<consul-token>" -X PUT "http://localhost:8500/v1/kv/edgex/v3/device-onvif-camera/AppCustom/CredentialsMap/<creds-name>"
+    ```
```

Comment:
I would have setting the secret be the first item in these instructions. Also, do not use these commands to set the credentials. you want to use the command `/api/v3/secret`.

Reply:
Oops forgot about that part. I was just following the flow of the scripts and was going to change it back.